src/vs/editor/contrib/gotoSymbol/browser/goToCommands.ts

```diff
@@ -249,6 +249,33 @@ export abstract class SymbolNavigationAction extends EditorAction2 {
 	}
 }
 
+export class KnownTargetNavigationAction extends SymbolNavigationAction {
```

Comment:
Not really sure why this was added. Please explain or remove to keep the PR minimal and focused on the actual issue

Reply:
Previously, `DefinitionAction` was used for inlay hints, and `DefinitionAction` always goes to the definition of token under cursor. This one is similar to `DefinitionAction` and other classes in this category, but it gets the target of navigation in its constructor. It is used instead of `DefinitionAction` in `inlayHintsLocation.ts`.

Reply:
> but it gets the target of navigation in its constructor. It is used instead of DefinitionAction in inlayHintsLocation.ts.

Careful with that. While `InlayHintLabelPart#location` looks like (and usually is) the definition of the symbol it must not be that. The contract is to give a location at which calling go-to-def is doing the right thing. The easiest way to think about this is symbols with multiple definition locations. We don't want that to surface in the inlay hint label but still want to do the right thing. That'd why we cannot take it as result/target but only as anchor (for a result)

Reply:
Thanks for explanation. I now understand why it was the way it was, and so I found out that my changes were unnecessary and a small tweak was enough to make it working. `KnownTargetNavigationAction` is now removed.